pkg/services/notification/controller.go

```diff
@@ -24,21 +24,23 @@ import (
 type Controller struct {
 	runningTaskIds         chan string
 	runningNotificationIds chan string
-	taskQueue              []*etcd.Queue
-	notificationQueue      []*etcd.Queue
+	taskQueue              lib.Topic
+	notificationQueue      lib.Topic
 }
 
 func NewController() *Controller {
-	queueNum := etcd.GetQueueNum()
-
-	notificationQueue := make([]*etcd.Queue, 0)
-	taskQueue := make([]*etcd.Queue, 0)
-	e := global.GetInstance().GetEtcd()
-	for i := 0; i < queueNum; i++ {
-		notificationQueue = append(notificationQueue, e.NewQueue(fmt.Sprintf("%s-%d", constants.NotificationTopicPrefix, i)))
-		taskQueue = append(taskQueue, e.NewQueue(fmt.Sprintf("%s-%d", constants.NotificationTaskTopicPrefix, i)))
+	cfg := config.GetInstance().LoadConf()
+	redisConnStr := cfg.Redis.Addr
+	c, err := r.New(redisConnStr)
```

Comment:
New queue by queue.type, here should no care about which package will be import.

```
delete r "openpitrix.io/libqueue/redisqueue"

cfg := config.GetInstance().LoadConf()

queueAddr := cfg.Queue.Addr
queueType := cfg.Queue.Type
c, err := r.New(queueType, queueAddr)
or 
c, err := r.New(&someQueueStruct{type: queueType, addr: queueAddr})
```